Add polling parameters to ConfigManagingActor

daniel-zullo-frequenz opened the pull request:
The ConfigManagingActor now allows to force file polling and set the polling interval.

daniel-zullo-frequenz committed:
Add polling parameters to ConfigManagingActor

The ConfigManagingActor now allows to force file polling
and set the polling interval.

Signed-off-by: Daniel Zullo <daniel.zullo@frequenz.com>

llucax commented:
Damn, cross-arch tests are consistently failing for `pytest_max` when adding this to the merge queue.

```
FAILED tests/timeseries/_battery_pool/test_battery_pool.py::test_battery_pool_power_bounds - assert 0.855949 < 0.22
 +  where 0.855949 = <built-in method total_seconds of datetime.timedelta object at 0x5512d00630>()
 +    where <built-in method total_seconds of datetime.timedelta object at 0x5512d00630> = datetime.timedelta(microseconds=855949).total_seconds
```

Super weird, I can't see ow this change could be related to this.

shsms committed:
Reduce wait time in battery_pool test

This test was failing in arm64 runs, because of a large time
difference.  This fixes the issue.

Signed-off-by: Sahas Subramanian <sahas.subramanian@proton.me>

shsms commented:
I fixed it, in the latest commit.  Here's a successful run: https://github.com/shsms/frequenz-sdk-python/actions/runs/11201255580/job/31135960350

It is too complicated to explain how this fixes it, so I won't.

llucax commented:
> It is too complicated to explain how this fixes it, so I won't.

This is not great :worried: 

I see in the commit with the fix you at least explain something. Can you maybe mention at least what do you think it triggers this issue, as it seems completely unrelated to the update to of the channels version? I wonder if there is something we can learn from here about expecting similar issues when upgrading other projects.

llucax reviewed (approved):
So the changes LGTM, I would like to know more about the root cause of the new breakage, in case we bump into similar issues when upgrading to channels 1.2 in other projects, but we can merge anyways.

shsms commented:
> I see in the commit with the fix you at least explain something. Can you maybe mention at least what do you think it triggers this issue, as it seems completely unrelated to the update to of the channels version? I wonder if there is something we can learn from here about expecting similar issues when upgrading other projects.

I actually have no idea.  The commit reduces the delay by 0.1 seconds, so technically it couldn't have made up for the 0.6s lag that we were seeing in the failing test.  I think it is just software emulated VM flakiness.

shsms commented:
> It is too complicated to explain how this fixes it, so I won't.

That was actually a joke, I guess it wasn't obvious. :D

llucax commented:
> That was actually a joke, I guess it wasn't obvious. :D

No, I'm too scared of this kind of failure/flakiness, I guess my jokes detector doesn't perform well in these kind of extreme conditions :P
